### PR TITLE
[pt2] Log is_forward field to dynamo_compile scuba table

### PR DESCRIPTION
**Pull request by masnesral**
Summary: ^^

Test Plan:
Ran a test script out of fbcode: D64350202. Then:

```
(pytorch-3.10_4) devvm2296:~/fbcode  $ scuba -e="select time,co_filename,is_forward from \`dynamo_compile/sandbox\` where is_forward is not null"
+------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
|    time    |                                                                                    co_filename                                                                                    | is_forward |
+------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| 1729032583 | /data/users/slarsen/fbsource/buck-out/v2/gen/fbcode/1638b36e975169f6/scripts/slarsen/torch_compile_model/__run__/run-inplace#link-tree/scripts/slarsen/torch_compile_model/run.py |          1 |
| 1729032583 | null                                                                                                                                                                              |          0 |
| 1729032650 | /data/users/slarsen/fbsource/buck-out/v2/gen/fbcode/1638b36e975169f6/scripts/slarsen/torch_compile_model/__run__/run-inplace#link-tree/scripts/slarsen/torch_compile_model/run.py |          1 |
| 1729032650 | null                                                                                                                                                                              |          0 |
+------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
4 row(s) in set (0 warnings, 131 errors, 0.80 sec)
```

Reviewed By: ezyang

Differential Revision: D64438144

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -814,6 +815,7 @@ class CompilationMetrics:
 
 @dataclasses.dataclass
 class BwdCompilationMetrics:
+    is_forward: bool = dataclasses.field(default=False, init=False)
```
Comment (ezyang) [inline torch/_dynamo/utils.py:818]: By the way it's so weird to me that you put this at the top of the dataclass, it feels very BC breaking to do it that way if people are passing in arguments positionally lol

Comment (ezyang) [reply torch/_dynamo/utils.py:818]: like if tests pass I guess it's fine it's just weird lol

Comment (masnesral) [reply torch/_dynamo/utils.py:818]: @ezyang, yeah I normally wouldn't have, but I tried to make this field essentially a static property of the class -- not provided at initialization. So somehow it made more sense to me to stick it at the top. The next time someone appends a new field to this dataclass, I'm hoping they won't be confused about why is_forward is not passed.

**Comment by masnesral** [comment]
@huydhn what do you mean? This only touches `torch/_dynamo/utils.py`, and I don't see that file modified by https://github.com/pytorch/pytorch/pull/137803

**Comment by huydhn** [comment]
Let me add `ci-no-td` to ensure that target determination is not enabled on the PR, and all tests are run

**Pull request closed without merging**